src/main/java/org/sunbird/assessment/service/AssessmentServiceV2Impl.java

```diff
@@ -34,146 +39,329 @@ public class AssessmentServiceV2Impl implements AssessmentServiceV2 {
-                            .get(Constants.RESULT)).get(Constants.QUESTIONS));
-                    for (Map<String, Object> qmap : questionMap) {
-                        if (!ObjectUtils.isEmpty(questionMap)) {
-                            redisCacheMgr.putCache(Constants.QUESTION_ID + qmap.get(Constants.IDENTIFIER), qmap);
-                            questionList.add(filterQuestionMapDetail(qmap));
-                        } else {
-                            logger.error(new Exception("Failed to get Question Details for Id: " + qmap.get(Constants.IDENTIFIER)));
+            String userId = RequestInterceptor.fetchUserIdFromAccessToken(authUserToken);
+            if (userId != null) {
+                List<String> identifierList = getQuestionIdList(requestBody);
+                List<Object> questionList = new ArrayList<>();
+                List<String> newIdentifierList = new ArrayList<>();
+                if (requestBody.containsKey(Constants.ASSESSMENT_ID_KEY) && !requestBody.get(Constants.ASSESSMENT_ID_KEY).toString().isEmpty() && !identifierList.isEmpty()) {
+                    String key = Constants.USER_ASSESS_REQ + requestBody.get(Constants.ASSESSMENT_ID_KEY).toString() + authUserToken;
+                    Map<String, Object> questionSetFromAssessment = (Map<String, Object>) redisCacheMgr.getCache(key);
+                    if (questionSetFromAssessment != null && questionSetFromAssessment.get(Constants.CHILDREN) != null) {
+                        List<String> questionsFromAssessment = new ArrayList<>();
+                        List<Map<String, Object>> sections = (List<Map<String, Object>>) questionSetFromAssessment.get(Constants.CHILDREN);
+                        for (Map<String, Object> section : sections) {
+                            questionsFromAssessment.addAll((List<String>) section.get(Constants.CHILD_NODES));
+                        }
+                        //Out of the list of questions received in the payload, retaining only those ids which are a part of the user's latest assessment
+                        identifierList.retainAll(questionsFromAssessment);
+                        //Fetching all the remaining questions details from the Redis
+                        List<Object> map = redisCacheMgr.mget(identifierList);
+                        for (int i = 0; i < map.size(); i++) {
+                            if (ObjectUtils.isEmpty(map.get(i))) {
+                                //Adding the not found keys as a seperate list
+                                newIdentifierList.add(identifierList.get(i));
+                            } else {
+                                //Filtering the details of the questions which are found in redis and adding them to another list
+                                questionList.add(filterQuestionMapDetail((Map<String, Object>) map.get(i)));
+                            }
```

Comment:
yes, It returns null karthik, to preserve the order of the elements, otherwise anyone would get confused, so element 5 will have value in 5 as the response